// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.



// Remember to always delete the pb2 and pb2_grpc file and compile everytime something is added in proto file, use code below to compile,
//  python -m grpc_tools.protoc -I ../Safe_Entry --python_out=. --grpc_python_out=. ../Safe_Entry/safe_entry.proto

syntax = "proto3";

package Safe_Entry;


// The safe entry service definition.
service Safe_Entry {
  rpc Check_In (Check_In_Request) returns (Check_In_Reply) {}
}

// Contain the user information
message Persons {
    string user_id = 1;
    string location = 2;
    string date = 3;
    string time = 4;
}

// The request message containing the Check-In Request Information.
message Check_In_Request {
    string user_id = 1;
    string location = 2;
    string date = 3;
    string time = 4;

}

// The response message containing the Check-In Reply Information
message Check_In_Reply {
  string user_id_res = 1;
  string location_res = 2;
  string date_res = 3;
  string time_res = 4;
}
